Legion/VaultOfTheWardens/Tormentorum.lua

```diff
@@ -115,3 +123,63 @@ end
 function mod:FrighteningShout(args)
 	self:Message(args.spellId, "Urgent", "Warning", CL.casting:format(args.spellName))
 end
+
+do
+	local prev = 0
+	function mod:ShadowCrashDamage(args)
+		local t = GetTime()
+		if self:Me(args.destGUID) and t-prev > 1.5 then
+			prev = t
+			self:Message(args.spellId, "Personal", "Alert", CL.underyou:format(args.spellName))
+		end
+	end
+end
+
+do
+	local prev = 0
+
+	local function printStacks(self, key)
+		local meOnly = self:CheckOption(key, "ME_ONLY")
+		if meOnly then
+			local stacks = select(4, UnitDebuff("player", self:SpellName(key))) or 0
+			if stacks > 6 then
+				self:StackMessage(key, UnitName("player"), stacks, "Urgent", stacks < 9 and "Alarm" or "Warning")
+			end
+		else
+			local highestStacks, stacksOnMe = 0, 0
+			local affectedPlayers = {}
+			for unit in self:IterateGroup() do
+				local stacks = select(4, UnitDebuff(unit, self:SpellName(key))) or 0
+				if stacks > 6 then
+					affectedPlayers[#affectedPlayers + 1] = { name = UnitName(unit), stacks = stacks }
+					if highestStacks < stacks then
+						highestStacks = stacks
+					end
+					if UnitIsUnit("player", unit) then
+						stacksOnMe = stacks
+					end
+				end
+			end
+			if #affectedPlayers == 1 then
+				self:StackMessage(key, affectedPlayers[1].name, affectedPlayers[1].stacks, "Urgent", (UnitName("player") == affectedPlayers[1].name or self:Dispeller("magic")) and (affectedPlayers[1].stacks < 9 and "Alarm" or "Warning"))
+			else
+				for i, v in ipairs(list) do

```

Comment:
The list should be sorted before this, highest stacks first.
This will also eliminate the need of `highestStacks`, since that's always the first table entry.